### DISCUSSION TITLE
What is purpose of using ThemeProvider?

### DISCUSSION THREAD

**Discussion by hoffmannkrzysztof** [EXERLOG/exer_log#151] [General]
What is a reason of using custom ThemeProvider instead of using native light/dart mode in MaterialApp? We should migrate all colors to separate classes ThemeDate/

**Comment by hoffmannkrzysztof** [comment]
I prepare PR for this. Take a look and see what i mean. https://github.com/EXERLOG/exer_log/pull/153

**Reply by momshaddinury** [reply]
So your idea is to assign colors to `theme` directly and then leverage `Theme.of(context)` everywhere. It's not bad. @jorre127 what do you think?

**Reply by jorre127** [reply]
@hoffmannkrzysztof It's not a bad idea to use the theme from the materiaApp but I would use it in combination with the current implementation. Because now we can define our own UI tokens by having our own implementation which gives us way more flexibility, the plan was also to house the theme textstyles in the AppTheme so they also support multiple themes, so I would definitely keep the current implementation. But also add some colors to theme of the material app because they give some item standard styles which is nice so you don't have to define them custom every time.

**Reply by jorre127** [reply]
its especially handy when we have a design because we can map our UI tokens one to one with the ones used in the design.

**Reply by hoffmannkrzysztof** [reply]
Understood, i thought that you will using only light/dart theme. If you want to use other themes that solution make sense.

**Reply by jorre127** [reply]
It’s not really decided yet if we would have multiple themes. But the fact we can is one of the bonuses
